__tests__/unit/result.test.ts

```diff
@@ -0,0 +1,74 @@
+import type { Row } from '../../src'
+import { ResultSet } from '../../src'
+import Stream, { Readable } from 'stream'
+
+describe('rows', () => {
+  const expectedText = `{"foo":"bar"}\n{"qaz":"qux"}\n`
+  const expectedJson = [{ foo: 'bar' }, { qaz: 'qux' }]
+
+  const err = 'Stream has been already consumed'
+
+  it('should consume the response as text only once', async () => {
+    const rs = makeResultSet()
+
+    expect(await rs.text()).toEqual(expectedText)
+    await expect(rs.text()).rejects.toThrowError(err)
+    await expect(rs.json()).rejects.toThrowError(err)
+  })
+
+  it('should consume the response as JSON only once', async () => {
+    const rs = makeResultSet()
+
+    expect(await rs.json()).toEqual(expectedJson)
+    await expect(rs.json()).rejects.toThrowError(err)
+    await expect(rs.text()).rejects.toThrowError(err)
+  })
+
+  it('should consume the response as a stream of Row instances', async () => {
+    const rs = makeResultSet()
+    const stream = rs.stream()
+
+    expect(stream.readableEnded).toBeFalsy()
+
+    const result: unknown[] = []
+    for await (const rows of rs.stream()) {
+      rows.forEach((row: Row) => {
+        result.push(row.json())
+      })
+    }
+
+    expect(result).toEqual(expectedJson)
+    // expect(stream.readableEnded).toBeTruthy()
```

Comment:
that's a bit strange, but it does not have `readableEnded` set to `true` anymore. I wonder if we can fix it.